### ISSUE TITLE
Spontaneous environment error / import error causing "CommandError: Unknown command: 'migrate'"

### ISSUE THREAD

**Issue by benjaoming** [learningequality/ka-lite#4386]
Branch: 0.14.x, latest

```
➜  ka-lite git:(fix-tests) ✗ kalite manage setup      
                                     
   _   __  ___    _     _ _          
  | | / / / _ \  | |   (_) |         
  | |/ / / /_\ \ | |    _| |_ ___    
  |    \ |  _  | | |   | | __/ _ \   
  | |\  \| | | | | |___| | ||  __/   
  \_| \_/\_| |_/ \_____/_|\__\___|   
                                     
https://learningequality.org/ka-lite/
                                     
         version 0.14.0
                                     
WARNING:root:It's recommended that you install Python version 2.7.9. Your version is: 2.7.6

--------------------------------------------------------------------------------
This script will configure the database and prepare it for use.
--------------------------------------------------------------------------------
Press [enter] to continue...
-------------------------------------------------------------------
WARNING: Database file already exists!
-------------------------------------------------------------------
Keep database file and upgrade to KA Lite version 0.14.0? (yes or no) yes
Upgrading database to KA Lite version 0.14.0
Cleaning *pyc files (if writable) from: ka-lite
Creating tables ...
Installing custom SQL ...
Installing indexes ...
Installed 0 object(s) from 0 fixture(s)
Traceback (most recent call last):
  File "ka-lite/python-packages/django/core/management/base.py", line 224, in run_from_argv
    self.execute(*args, **options.__dict__)
  File "ka-lite/python-packages/django/core/management/base.py", line 263, in execute
    output = self.handle(*args, **options)
  File "ka-lite/kalite/distributed/management/commands/setup.py", line 370, in handle
    call_command("migrate", merge=True, verbosity=options.get("verbosity"))
  File "ka-lite/python-packages/django/core/management/__init__.py", line 141, in call_command
    raise CommandError("Unknown command: %r" % name)
CommandError: Unknown command: 'migrate'
➜  ka-lite git:(fix-tests) ✗ pip install -r requirements.txt 
You are using pip version 6.0.7, however version 7.1.2 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
Requirement already satisfied (use --upgrade to upgrade): django-compressor==1.4 in /home/benjamin/.virtualenvs/kalite/lib/python2.7/site-packages (from -r requirements.txt (line 1))
Requirement already satisfied (use --upgrade to upgrade): docopt<0.7,>=0.6 in /home/benjamin/.virtualenvs/kalite/lib/python2.7/site-packages (from -r requirements.txt (line 2))
Requirement already satisfied (use --upgrade to upgrade): South==0.8.4 in /home/benjamin/.virtualenvs/kalite/lib/python2.7/site-packages (from -r requirements.txt (line 3))
Requirement already satisfied (use --upgrade to upgrade): django-js-reverse==0.5.0 in /home/benjamin/.virtualenvs/kalite/lib/python2.7/site-packages (from -r requirements.txt (line 4))
Requirement already satisfied (use --upgrade to upgrade): ply==3.4 in /home/benjamin/.virtualenvs/kalite/lib/python2.7/site-packages (from -r requirements.txt (line 6))
Requirement already satisfied (use --upgrade to upgrade): slimit==0.8.1 in /home/benjamin/.virtualenvs/kalite/lib/python2.7/site-packages (from -r requirements.txt (line 8))
Requirement already satisfied (use --upgrade to upgrade): six==1.9.0 in /home/benjamin/.virtualenvs/kalite/lib/python2.7/site-packages (from -r requirements.txt (line 9))
Requirement already satisfied (use --upgrade to upgrade): django-appconf==1.0.1 in /home/benjamin/.virtualenvs/kalite/lib/python2.7/site-packages (from -r requirements.txt (line 10))
Requirement already satisfied (use --upgrade to upgrade): Django>=1.5 in /home/benjamin/.virtualenvs/kalite/lib/python2.7/site-packages (from django-js-reverse==0.5.0->-r requirements.txt (line 4))
➜  ka-lite git:(fix-tests) ✗ kalite manage setup            
                                     
   _   __  ___    _     _ _          
  | | / / / _ \  | |   (_) |         
  | |/ / / /_\ \ | |    _| |_ ___    
  |    \ |  _  | | |   | | __/ _ \   
  | |\  \| | | | | |___| | ||  __/   
  \_| \_/\_| |_/ \_____/_|\__\___|   
                                     
https://learningequality.org/ka-lite/
                                     
         version 0.14.0
                                     
WARNING:root:It's recommended that you install Python version 2.7.9. Your version is: 2.7.6

--------------------------------------------------------------------------------
This script will configure the database and prepare it for use.
--------------------------------------------------------------------------------
Press [enter] to continue...
-------------------------------------------------------------------
WARNING: Database file already exists!
-------------------------------------------------------------------
Keep database file and upgrade to KA Lite version 0.14.0? (yes or no) yes
Upgrading database to KA Lite version 0.14.0
Cleaning *pyc files (if writable) from: ka-lite
Syncing...
Creating tables ...
Installing custom SQL ...
Installing indexes ...
Installed 0 object(s) from 0 fixture(s)

Synced:
 > django.contrib.auth
 > django.contrib.sessions
 > django.contrib.messages
 > django.contrib.admin
 > django.contrib.staticfiles
 > django.contrib.contenttypes
 > compressor
 > django_js_reverse
 > south
 > fle_utils.testing
 > kalite.coachreports
 > kalite.distributed
 > kalite.topic_tools
 > kalite.contentload
 > kalite.dynamic_assets
 > kalite.control_panel
 > kalite.testing
 > kalite.testing.loadtesting
 > kalite.basetests

Not synced (use migrations):
 - securesync
 - fle_utils.config
 - fle_utils.chronograph
 - kalite.main
 - kalite.playlist
 - kalite.updates
 - kalite.facility
 - kalite.student_testing
 - kalite.store
 - kalite.i18n
 - tastypie
(use ./manage.py migrate to migrate these)
Running migrations for securesync:
- Nothing to migrate.
 - Loading initial data for securesync.
Installed 0 object(s) from 0 fixture(s)
Running migrations for config:
- Nothing to migrate.
 - Loading initial data for config.
Installed 0 object(s) from 0 fixture(s)
Running migrations for chronograph:
- Nothing to migrate.
 - Loading initial data for chronograph.
Installed 0 object(s) from 0 fixture(s)
Running migrations for main:
- Nothing to migrate.
 - Loading initial data for main.
Installed 0 object(s) from 0 fixture(s)
Running migrations for playlist:
- Nothing to migrate.
 - Loading initial data for playlist.
Installed 0 object(s) from 0 fixture(s)
Running migrations for updates:
- Nothing to migrate.
 - Loading initial data for updates.
Installed 0 object(s) from 0 fixture(s)
Running migrations for facility:
- Nothing to migrate.
 - Loading initial data for facility.
Installed 0 object(s) from 0 fixture(s)
Running migrations for student_testing:
- Nothing to migrate.
 - Loading initial data for student_testing.
Installed 0 object(s) from 0 fixture(s)
Running migrations for store:
- Nothing to migrate.
 - Loading initial data for store.
Installed 0 object(s) from 0 fixture(s)
Running migrations for i18n:
- Nothing to migrate.
 - Loading initial data for i18n.
Installed 0 object(s) from 0 fixture(s)
Running migrations for tastypie:
- Nothing to migrate.
 - Loading initial data for tastypie.
Installed 0 object(s) from 0 fixture(s)
Syncing...
Creating tables ...
Installing custom SQL ...
Installing indexes ...
Installed 0 object(s) from 0 fixture(s)

Synced:
 > django.contrib.auth
 > django.contrib.sessions
 > django.contrib.messages
 > django.contrib.admin
 > django.contrib.staticfiles
 > django.contrib.contenttypes
 > compressor
 > django_js_reverse
 > south
 > fle_utils.testing
 > kalite.coachreports
 > kalite.distributed
 > kalite.topic_tools
 > kalite.contentload
 > kalite.dynamic_assets
 > kalite.control_panel
 > kalite.testing
 > kalite.testing.loadtesting
 > kalite.basetests

Not synced (use migrations):
 - securesync
 - fle_utils.config
 - fle_utils.chronograph
 - kalite.main
 - kalite.playlist
 - kalite.updates
 - kalite.facility
 - kalite.student_testing
 - kalite.store
 - kalite.i18n
 - tastypie
(use ./manage.py migrate to migrate these)

Starting in version 0.13, you will need an assessment items package in order to access many of the available exercises.
If you have an internet connection, you can download the needed package. Warning: this may take a long time!
If you have already downloaded the assessment items package, you can specify the file in the next step.
Otherwise, we will download it from https://learningequality.org/downloads/ka-lite/0.14/content/assessment.zip.
Do you wish to download the assessment items package now? (yes or no) 

...
```

What happens:

 1. I run `kalite manage setup` it fails, saying command `migrate` does not exist
 1. I run `pip install requirements.txt` - nothing is missing!
 1. I run `kalite manage setup` again and it actually works.

I'm suspecting that we have some kind of load time race condition that can spontaneously cause circularity or some other kind of import issue that pops up once in a while.

Let's track the issue here and see what happens.

**Comment by aronasorman** [comment]
0.14.x is now in maintenance mode. This isn't critical, so I don't think this should be for 0.14. Let's reopen this if this is still extant on develop.

**Comment by benjaoming** [comment]
Please don't close it. It hasn't been fixed.

**Comment by benjaoming** [comment]
If we find the cause, we should backport it as much as possible.

**Comment by rtibbles** [comment]
Is this still extant, @benjaoming ?

**Comment by benjaoming** [comment]
I think that once #3952 is solved and we stop deleting *pyc files, we can try closing this... but I'm not too sure we should consider it a "solved" issue.

**Comment by benjaoming** [comment]
Fixed in #5285 - assuming this was a spontaneuous issue of *pyc files being created/removed during runtime. `kalite manage setup` no longer deletes *pyc because it's all assumed handled by setuptools and installers.
